### PR TITLE
[BD-46] docs: add click analytics for Algolia search

### PR DESCRIPTION
**Pull request by viktorrusakov**
## Description

Send click event data to Algolia after successful search.

### Deploy Preview

Unfortunately search functionality does not work in deploy previews 🥲 

## Merge Checklist

* [x] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [x] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [x] Were your changes tested in the `example` app?
* [x] Is there adequate test coverage for your changes?
* [x] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @viktorrusakov!


When this pull request is ready, tag your edX technical lead.

File: www/src/components/Search/index.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      hitComponent={({ hit, children }) => <HitComponent userId={userId} hit={hit}>{children}</HitComponent>}
+      searchParameters={{
+        clickAnalytics: true,
+      }}
+      // DocSearch component has poor support for Algolia's click analytics, that's why we need to
+      // 1. Modify searchClient behaviour to get queryID from search response in order to send it with click event data.
+      //    queryID is stored in localStorage instead of component state because changing state resets Search...
+      // 2. Use transformItems prop to include each item's position in the search results array,
+      //    these objects are then passed to hitComponent defined above through 'hit' prop, and then clicked
+      //    item's position will get sent to algolia with other click event data (it's a required parameter for the API)
```
Comment (adamstankiewicz) [inline www/src/components/Search/index.jsx:39]: 🙇 Helpful comment!

**Review by adamstankiewicz — APPROVED** [review]
LGTM. I am seeing events come through in the Algolia index's analytics for click-thru rate.

**Comment by openedx-webhooks** [comment]
@viktorrusakov 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1445&entry.1484655318=https://github.com/viktorrusakov&entry.752974735=2022-07-06+05:42&entry.1917517419=2022-07-08+10:05&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**